### PR TITLE
Adds signup() to Auth to default UI to a signup tab/form when available

### PR DESCRIPTION
**Pull request by dthyresson**
Addresses issue: https://github.com/redwoodjs/redwood/issues/963

Adds:

* signup() to AuthProvider
* Implements for Netlify and Auth0 dedicated signup flow
* For Firebase, the only provider supported is GoogleAuth and the signup and login are the same
* For Magic.link, login and signup are the same since it emails you a link to authenticate
* GoTrue uses signup API

Note: Firebase will eventually need to support specifying he provider (Twitter, Apple, Facebook, Firebase, etc). But then the login and signup will also have to handle different cases and even scopes.

Also, for Auth0, if you want to use the useAuth hook `Sign Up` to default the UI to the sign up "tab", you need to be using the ["New Universal Login Experience"](https://auth0.com/docs/universal-login/new-experience). The "Classic Universal Experience" does not support the `screen_hint` to set the tab.

Auth Playground PR exists to test once PR is available in a canary release.

### CODE REVIEW

**Comment by peterp** [comment]
Hmmmm... I wonder what happened with the commits here.

File: packages/auth/src/authClients/index.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export interface AuthClient {
   restoreAuthState?(): void | Promise<any>
   login(options?: any): Promise<any>
   logout(options?: any): void | Promise<void>
+  signup(options?: any): void | Promise<void> | Promise<any>
```
Comment (peterp) [inline packages/auth/src/authClients/index.ts:49]:
```suggestion
  signup(options?: any): void | Promise<any>
```

File: packages/auth/src/authClients/magicLink.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ export const magicLink = (client: MagicLink): AuthClientMagicLink => {
     logout: async () => {
       await client.user.logout()
     },
+    signup: async ({ email, showUI }) =>
+      await client.auth.loginWithMagicLink({ email: email, showUI: showUI }),
```
Comment (peterp) [inline packages/auth/src/authClients/magicLink.ts:22]:
```suggestion
      await client.auth.loginWithMagicLink({ email, showUI }),
```

**Review by peterp — APPROVED** [review]
LGTM

File: packages/auth/src/authClients/index.ts (hunk 1 above)
Comment (dthyresson) [reply packages/auth/src/authClients/index.ts:49]: Of course the any case covers void as well

File: packages/auth/src/authClients/magicLink.ts (hunk 2 above)
Comment (dthyresson) [reply packages/auth/src/authClients/magicLink.ts:22]: The `loginWithMagicLink` is the same as `login` so I will change both `login` and `signup` here.

**Comment by peterp** [comment]
Whoop whoop thanks @dthyresson !

**Pull request merged**